release fom PyPI doesn't contain all data needed to run unit tests

looks like a `setup.py` packaging error; will try to send fix today
```
$ dockerpy-pypi-tests wget http://pypi.python.org/packages/source/d/docker-py/docker-py-1.7.2.tar.gz
$ dockerpy-pypi-tests tar -xvf ./docker-py-1.7.2.tar.gz
docker-py-1.7.2/
docker-py-1.7.2/MANIFEST.in
docker-py-1.7.2/test-requirements.txt
docker-py-1.7.2/README.md
docker-py-1.7.2/setup.py
docker-py-1.7.2/PKG-INFO
docker-py-1.7.2/setup.cfg
docker-py-1.7.2/requirements.txt
docker-py-1.7.2/docker_py.egg-info/
docker-py-1.7.2/docker_py.egg-info/top_level.txt
docker-py-1.7.2/docker_py.egg-info/pbr.json
docker-py-1.7.2/docker_py.egg-info/requires.txt
docker-py-1.7.2/docker_py.egg-info/dependency_links.txt
docker-py-1.7.2/docker_py.egg-info/PKG-INFO
docker-py-1.7.2/docker_py.egg-info/not-zip-safe
docker-py-1.7.2/docker_py.egg-info/SOURCES.txt
docker-py-1.7.2/LICENSE
docker-py-1.7.2/tests/
docker-py-1.7.2/tests/integration/
docker-py-1.7.2/tests/integration/regression_test.py
docker-py-1.7.2/tests/integration/network_test.py
docker-py-1.7.2/tests/integration/volume_test.py
docker-py-1.7.2/tests/integration/image_test.py
docker-py-1.7.2/tests/integration/container_test.py
docker-py-1.7.2/tests/integration/conftest.py
docker-py-1.7.2/tests/integration/exec_test.py
docker-py-1.7.2/tests/integration/build_test.py
docker-py-1.7.2/tests/integration/__init__.py
docker-py-1.7.2/tests/integration/api_test.py
docker-py-1.7.2/tests/helpers.py
docker-py-1.7.2/tests/unit/
docker-py-1.7.2/tests/unit/fake_stat.py
docker-py-1.7.2/tests/unit/network_test.py
docker-py-1.7.2/tests/unit/volume_test.py
docker-py-1.7.2/tests/unit/image_test.py
docker-py-1.7.2/tests/unit/fake_api.py
docker-py-1.7.2/tests/unit/container_test.py
docker-py-1.7.2/tests/unit/auth_test.py
docker-py-1.7.2/tests/unit/exec_test.py
docker-py-1.7.2/tests/unit/build_test.py
docker-py-1.7.2/tests/unit/__init__.py
docker-py-1.7.2/tests/unit/utils_test.py
docker-py-1.7.2/tests/unit/api_test.py
docker-py-1.7.2/tests/base.py
docker-py-1.7.2/tests/__init__.py
docker-py-1.7.2/docker/
docker-py-1.7.2/docker/ssladapter/
docker-py-1.7.2/docker/ssladapter/ssladapter.py
docker-py-1.7.2/docker/ssladapter/__init__.py
docker-py-1.7.2/docker/tls.py
docker-py-1.7.2/docker/utils/
docker-py-1.7.2/docker/utils/utils.py
docker-py-1.7.2/docker/utils/decorators.py
docker-py-1.7.2/docker/utils/ports/
docker-py-1.7.2/docker/utils/ports/ports.py
docker-py-1.7.2/docker/utils/ports/__init__.py
docker-py-1.7.2/docker/utils/types.py
docker-py-1.7.2/docker/utils/__init__.py
docker-py-1.7.2/docker/errors.py
docker-py-1.7.2/docker/auth/
docker-py-1.7.2/docker/auth/auth.py
docker-py-1.7.2/docker/auth/__init__.py
docker-py-1.7.2/docker/api/
docker-py-1.7.2/docker/api/volume.py
docker-py-1.7.2/docker/api/build.py
docker-py-1.7.2/docker/api/exec_api.py
docker-py-1.7.2/docker/api/container.py
docker-py-1.7.2/docker/api/image.py
docker-py-1.7.2/docker/api/network.py
docker-py-1.7.2/docker/api/__init__.py
docker-py-1.7.2/docker/api/daemon.py
docker-py-1.7.2/docker/client.py
docker-py-1.7.2/docker/__init__.py
docker-py-1.7.2/docker/version.py
docker-py-1.7.2/docker/unixconn/
docker-py-1.7.2/docker/unixconn/unixconn.py
docker-py-1.7.2/docker/unixconn/__init__.py
docker-py-1.7.2/docker/constants.py
$ dockerpy-pypi-tests cd docker-py-1.7.2
$ d/docker-py-1.7.2 py.test-2 tests/unit
==================================================================================== test session starts =====================================================================================
platform linux2 -- Python 2.7.11, pytest-2.8.7, py-1.4.31, pluggy-0.3.1
rootdir: /home/tt/t/dockerpy-pypi-tests/docker-py-1.7.2, inifile:
plugins: cov-2.2.0
collected 304 items

tests/unit/api_test.py ......................
tests/unit/auth_test.py .........................................
tests/unit/build_test.py .........
tests/unit/container_test.py .................s.....................................................................
tests/unit/exec_test.py ....
tests/unit/image_test.py ........................
tests/unit/network_test.py ......
tests/unit/utils_test.py .................FFFF...................................................................................
tests/unit/volume_test.py .......

========================================================================================== FAILURES ==========================================================================================
____________________________________________________________________ KwargsFromEnvTest.test_kwargs_from_env_no_cert_path _____________________________________________________________________

self = <tests.unit.utils_test.KwargsFromEnvTest testMethod=test_kwargs_from_env_no_cert_path>

    def test_kwargs_from_env_no_cert_path(self):
        try:
            temp_dir = tempfile.mkdtemp()
            cert_dir = os.path.join(temp_dir, '.docker')
>           shutil.copytree(TEST_CERT_DIR, cert_dir)

tests/unit/utils_test.py:249:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

src = '/home/tt/t/dockerpy-pypi-tests/docker-py-1.7.2/tests/unit/testdata/certs', dst = '/tmp/tmpZjWLv5/.docker', symlinks = False, ignore = None

    def copytree(src, dst, symlinks=False, ignore=None):
        """Recursively copy a directory tree using copy2().

        The destination directory must not already exist.
        If exception(s) occur, an Error is raised with a list of reasons.

        If the optional symlinks flag is true, symbolic links in the
        source tree result in symbolic links in the destination tree; if
        it is false, the contents of the files pointed to by symbolic
        links are copied.

        The optional ignore argument is a callable. If given, it
        is called with the `src` parameter, which is the directory
        being visited by copytree(), and `names` which is the list of
        `src` contents, as returned by os.listdir():

            callable(src, names) -> ignored_names

        Since copytree() is called recursively, the callable will be
        called once for each directory that is copied. It returns a
        list of names relative to the `src` directory that should
        not be copied.

        XXX Consider this example code rather than the ultimate tool.

        """
>       names = os.listdir(src)
E       OSError: [Errno 2] No such file or directory: '/home/tt/t/dockerpy-pypi-tests/docker-py-1.7.2/tests/unit/testdata/certs'

/usr/lib64/python2.7/shutil.py:171: OSError
_________________________________________________________________________ KwargsFromEnvTest.test_kwargs_from_env_tls _________________________________________________________________________

self = <tests.unit.utils_test.KwargsFromEnvTest testMethod=test_kwargs_from_env_tls>

    def test_kwargs_from_env_tls(self):
        os.environ.update(DOCKER_HOST='tcp://192.168.59.103:2376',
                          DOCKER_CERT_PATH=TEST_CERT_DIR,
                          DOCKER_TLS_VERIFY='1')
>       kwargs = kwargs_from_env(assert_hostname=False)

tests/unit/utils_test.py:187:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
docker/utils/utils.py:491: in kwargs_from_env
    assert_hostname=assert_hostname,
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <docker.tls.TLSConfig object at 0x7f241cc64a90>
client_cert = ('/home/tt/t/dockerpy-pypi-tests/docker-py-1.7.2/tests/unit/testdata/certs/cert.pem', '/home/tt/t/dockerpy-pypi-tests/docker-py-1.7.2/tests/unit/testdata/certs/key.pem')
ca_cert = '/home/tt/t/dockerpy-pypi-tests/docker-py-1.7.2/tests/unit/testdata/certs/ca.pem', verify = True, ssl_version = None, assert_hostname = False, assert_fingerprint = None

    def __init__(self, client_cert=None, ca_cert=None, verify=None,
                 ssl_version=None, assert_hostname=None,
                 assert_fingerprint=None):
        # Argument compatibility/mapping with
        # https://docs.docker.com/engine/articles/https/
        # This diverges from the Docker CLI in that users can specify 'tls'
        # here, but also disable any public/default CA pool verification by
        # leaving tls_verify=False

        self.ssl_version = ssl_version
        self.assert_hostname = assert_hostname
        self.assert_fingerprint = assert_fingerprint

        # "tls" and "tls_verify" must have both or neither cert/key files
        # In either case, Alert the user when both are expected, but any are
        # missing.

        if client_cert:
            try:
                tls_cert, tls_key = client_cert
            except ValueError:
                raise errors.TLSParameterError(
                    'client_config must be a tuple of'
                    ' (client certificate, key file)'
                )

            if not (tls_cert and tls_key) or (not os.path.isfile(tls_cert) or
               not os.path.isfile(tls_key)):
                raise errors.TLSParameterError(
>                   'Path to a certificate and key files must be provided'
                    ' through the client_config param'
                )
E               TLSParameterError: Path to a certificate and key files must be provided through the client_config param. TLS configurations should map the Docker CLI client configurations. See https://docs.docker.com/engine/articles/https/ for API details.

docker/tls.py:42: TLSParameterError
__________________________________________________________________ KwargsFromEnvTest.test_kwargs_from_env_tls_verify_false ___________________________________________________________________

self = <tests.unit.utils_test.KwargsFromEnvTest testMethod=test_kwargs_from_env_tls_verify_false>

    def test_kwargs_from_env_tls_verify_false(self):
        os.environ.update(DOCKER_HOST='tcp://192.168.59.103:2376',
                          DOCKER_CERT_PATH=TEST_CERT_DIR,
                          DOCKER_TLS_VERIFY='')
>       kwargs = kwargs_from_env(assert_hostname=True)

tests/unit/utils_test.py:206:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
docker/utils/utils.py:491: in kwargs_from_env
    assert_hostname=assert_hostname,
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <docker.tls.TLSConfig object at 0x7f241cb6f610>
client_cert = ('/home/tt/t/dockerpy-pypi-tests/docker-py-1.7.2/tests/unit/testdata/certs/cert.pem', '/home/tt/t/dockerpy-pypi-tests/docker-py-1.7.2/tests/unit/testdata/certs/key.pem')
ca_cert = '/home/tt/t/dockerpy-pypi-tests/docker-py-1.7.2/tests/unit/testdata/certs/ca.pem', verify = False, ssl_version = None, assert_hostname = True, assert_fingerprint = None

    def __init__(self, client_cert=None, ca_cert=None, verify=None,
                 ssl_version=None, assert_hostname=None,
                 assert_fingerprint=None):
        # Argument compatibility/mapping with
        # https://docs.docker.com/engine/articles/https/
        # This diverges from the Docker CLI in that users can specify 'tls'
        # here, but also disable any public/default CA pool verification by
        # leaving tls_verify=False

        self.ssl_version = ssl_version
        self.assert_hostname = assert_hostname
        self.assert_fingerprint = assert_fingerprint

        # "tls" and "tls_verify" must have both or neither cert/key files
        # In either case, Alert the user when both are expected, but any are
        # missing.

        if client_cert:
            try:
                tls_cert, tls_key = client_cert
            except ValueError:
                raise errors.TLSParameterError(
                    'client_config must be a tuple of'
                    ' (client certificate, key file)'
                )

            if not (tls_cert and tls_key) or (not os.path.isfile(tls_cert) or
               not os.path.isfile(tls_key)):
                raise errors.TLSParameterError(
>                   'Path to a certificate and key files must be provided'
                    ' through the client_config param'
                )
E               TLSParameterError: Path to a certificate and key files must be provided through the client_config param. TLS configurations should map the Docker CLI client configurations. See https://docs.docker.com/engine/articles/https/ for API details.

docker/tls.py:42: TLSParameterError
______________________________________________________________ KwargsFromEnvTest.test_kwargs_from_env_tls_verify_false_no_cert _______________________________________________________________

self = <tests.unit.utils_test.KwargsFromEnvTest testMethod=test_kwargs_from_env_tls_verify_false_no_cert>

    def test_kwargs_from_env_tls_verify_false_no_cert(self):
        temp_dir = tempfile.mkdtemp()
        cert_dir = os.path.join(temp_dir, '.docker')
>       shutil.copytree(TEST_CERT_DIR, cert_dir)

tests/unit/utils_test.py:224:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

src = '/home/tt/t/dockerpy-pypi-tests/docker-py-1.7.2/tests/unit/testdata/certs', dst = '/tmp/tmpz0aF7c/.docker', symlinks = False, ignore = None

    def copytree(src, dst, symlinks=False, ignore=None):
        """Recursively copy a directory tree using copy2().

        The destination directory must not already exist.
        If exception(s) occur, an Error is raised with a list of reasons.

        If the optional symlinks flag is true, symbolic links in the
        source tree result in symbolic links in the destination tree; if
        it is false, the contents of the files pointed to by symbolic
        links are copied.

        The optional ignore argument is a callable. If given, it
        is called with the `src` parameter, which is the directory
        being visited by copytree(), and `names` which is the list of
        `src` contents, as returned by os.listdir():

            callable(src, names) -> ignored_names

        Since copytree() is called recursively, the callable will be
        called once for each directory that is copied. It returns a
        list of names relative to the `src` directory that should
        not be copied.

        XXX Consider this example code rather than the ultimate tool.

        """
>       names = os.listdir(src)
E       OSError: [Errno 2] No such file or directory: '/home/tt/t/dockerpy-pypi-tests/docker-py-1.7.2/tests/unit/testdata/certs'

/usr/lib64/python2.7/shutil.py:171: OSError
====================================================================== 4 failed, 299 passed, 1 skipped in 0.80 seconds =======================================================================
$ d/docker-py-1.7.2 py.test-3 tests/unit                                                                                                                              1 ↵
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.5.1, pytest-2.8.7, py-1.4.31, pluggy-0.3.1
rootdir: /home/tt/t/dockerpy-pypi-tests/docker-py-1.7.2, inifile:
plugins: capturelog-0.7, cov-2.2.0
collected 304 items

tests/unit/api_test.py ......................
tests/unit/auth_test.py .........................................
tests/unit/build_test.py .........
tests/unit/container_test.py .................s.....................................................................
tests/unit/exec_test.py ....
tests/unit/image_test.py ........................
tests/unit/network_test.py ......
tests/unit/utils_test.py .................FFFF.................................s.................................................
tests/unit/volume_test.py .......

========================================================================================== FAILURES ==========================================================================================
____________________________________________________________________ KwargsFromEnvTest.test_kwargs_from_env_no_cert_path _____________________________________________________________________

self = <tests.unit.utils_test.KwargsFromEnvTest testMethod=test_kwargs_from_env_no_cert_path>

    def test_kwargs_from_env_no_cert_path(self):
        try:
            temp_dir = tempfile.mkdtemp()
            cert_dir = os.path.join(temp_dir, '.docker')
>           shutil.copytree(TEST_CERT_DIR, cert_dir)

tests/unit/utils_test.py:249:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

src = '/home/tt/t/dockerpy-pypi-tests/docker-py-1.7.2/tests/unit/testdata/certs', dst = '/tmp/tmp56f8je2e/.docker', symlinks = False, ignore = None
copy_function = <function copy2 at 0x7f26369a9ea0>, ignore_dangling_symlinks = False

    def copytree(src, dst, symlinks=False, ignore=None, copy_function=copy2,
                 ignore_dangling_symlinks=False):
        """Recursively copy a directory tree.

        The destination directory must not already exist.
        If exception(s) occur, an Error is raised with a list of reasons.

        If the optional symlinks flag is true, symbolic links in the
        source tree result in symbolic links in the destination tree; if
        it is false, the contents of the files pointed to by symbolic
        links are copied. If the file pointed by the symlink doesn't
        exist, an exception will be added in the list of errors raised in
        an Error exception at the end of the copy process.

        You can set the optional ignore_dangling_symlinks flag to true if you
        want to silence this exception. Notice that this has no effect on
        platforms that don't support os.symlink.

        The optional ignore argument is a callable. If given, it
        is called with the `src` parameter, which is the directory
        being visited by copytree(), and `names` which is the list of
        `src` contents, as returned by os.listdir():

            callable(src, names) -> ignored_names

        Since copytree() is called recursively, the callable will be
        called once for each directory that is copied. It returns a
        list of names relative to the `src` directory that should
        not be copied.

        The optional copy_function argument is a callable that will be used
        to copy each file. It will be called with the source path and the
        destination path as arguments. By default, copy2() is used, but any
        function that supports the same signature (like copy()) can be used.

        """
>       names = os.listdir(src)
E       FileNotFoundError: [Errno 2] No such file or directory: '/home/tt/t/dockerpy-pypi-tests/docker-py-1.7.2/tests/unit/testdata/certs'

/usr/lib64/python3.5/shutil.py:303: FileNotFoundError
_________________________________________________________________________ KwargsFromEnvTest.test_kwargs_from_env_tls _________________________________________________________________________

self = <tests.unit.utils_test.KwargsFromEnvTest testMethod=test_kwargs_from_env_tls>

    def test_kwargs_from_env_tls(self):
        os.environ.update(DOCKER_HOST='tcp://192.168.59.103:2376',
                          DOCKER_CERT_PATH=TEST_CERT_DIR,
                          DOCKER_TLS_VERIFY='1')
>       kwargs = kwargs_from_env(assert_hostname=False)

tests/unit/utils_test.py:187:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
docker/utils/utils.py:491: in kwargs_from_env
    assert_hostname=assert_hostname,
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <docker.tls.TLSConfig object at 0x7f262f5f6550>
client_cert = ('/home/tt/t/dockerpy-pypi-tests/docker-py-1.7.2/tests/unit/testdata/certs/cert.pem', '/home/tt/t/dockerpy-pypi-tests/docker-py-1.7.2/tests/unit/testdata/certs/key.pem')
ca_cert = '/home/tt/t/dockerpy-pypi-tests/docker-py-1.7.2/tests/unit/testdata/certs/ca.pem', verify = True, ssl_version = None, assert_hostname = False, assert_fingerprint = None

    def __init__(self, client_cert=None, ca_cert=None, verify=None,
                 ssl_version=None, assert_hostname=None,
                 assert_fingerprint=None):
        # Argument compatibility/mapping with
        # https://docs.docker.com/engine/articles/https/
        # This diverges from the Docker CLI in that users can specify 'tls'
        # here, but also disable any public/default CA pool verification by
        # leaving tls_verify=False

        self.ssl_version = ssl_version
        self.assert_hostname = assert_hostname
        self.assert_fingerprint = assert_fingerprint

        # "tls" and "tls_verify" must have both or neither cert/key files
        # In either case, Alert the user when both are expected, but any are
        # missing.

        if client_cert:
            try:
                tls_cert, tls_key = client_cert
            except ValueError:
                raise errors.TLSParameterError(
                    'client_config must be a tuple of'
                    ' (client certificate, key file)'
                )

            if not (tls_cert and tls_key) or (not os.path.isfile(tls_cert) or
               not os.path.isfile(tls_key)):
                raise errors.TLSParameterError(
>                   'Path to a certificate and key files must be provided'
                    ' through the client_config param'
                )
E               docker.errors.TLSParameterError: Path to a certificate and key files must be provided through the client_config param. TLS configurations should map the Docker CLI client configurations. See https://docs.docker.com/engine/articles/https/ for API details.

docker/tls.py:42: TLSParameterError
__________________________________________________________________ KwargsFromEnvTest.test_kwargs_from_env_tls_verify_false ___________________________________________________________________

self = <tests.unit.utils_test.KwargsFromEnvTest testMethod=test_kwargs_from_env_tls_verify_false>

    def test_kwargs_from_env_tls_verify_false(self):
        os.environ.update(DOCKER_HOST='tcp://192.168.59.103:2376',
                          DOCKER_CERT_PATH=TEST_CERT_DIR,
                          DOCKER_TLS_VERIFY='')
>       kwargs = kwargs_from_env(assert_hostname=True)

tests/unit/utils_test.py:206:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
docker/utils/utils.py:491: in kwargs_from_env
    assert_hostname=assert_hostname,
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <docker.tls.TLSConfig object at 0x7f262f5484e0>
client_cert = ('/home/tt/t/dockerpy-pypi-tests/docker-py-1.7.2/tests/unit/testdata/certs/cert.pem', '/home/tt/t/dockerpy-pypi-tests/docker-py-1.7.2/tests/unit/testdata/certs/key.pem')
ca_cert = '/home/tt/t/dockerpy-pypi-tests/docker-py-1.7.2/tests/unit/testdata/certs/ca.pem', verify = False, ssl_version = None, assert_hostname = True, assert_fingerprint = None

    def __init__(self, client_cert=None, ca_cert=None, verify=None,
                 ssl_version=None, assert_hostname=None,
                 assert_fingerprint=None):
        # Argument compatibility/mapping with
        # https://docs.docker.com/engine/articles/https/
        # This diverges from the Docker CLI in that users can specify 'tls'
        # here, but also disable any public/default CA pool verification by
        # leaving tls_verify=False

        self.ssl_version = ssl_version
        self.assert_hostname = assert_hostname
        self.assert_fingerprint = assert_fingerprint

        # "tls" and "tls_verify" must have both or neither cert/key files
        # In either case, Alert the user when both are expected, but any are
        # missing.

        if client_cert:
            try:
                tls_cert, tls_key = client_cert
            except ValueError:
                raise errors.TLSParameterError(
                    'client_config must be a tuple of'
                    ' (client certificate, key file)'
                )

            if not (tls_cert and tls_key) or (not os.path.isfile(tls_cert) or
               not os.path.isfile(tls_key)):
                raise errors.TLSParameterError(
>                   'Path to a certificate and key files must be provided'
                    ' through the client_config param'
                )
E               docker.errors.TLSParameterError: Path to a certificate and key files must be provided through the client_config param. TLS configurations should map the Docker CLI client configurations. See https://docs.docker.com/engine/articles/https/ for API details.

docker/tls.py:42: TLSParameterError
______________________________________________________________ KwargsFromEnvTest.test_kwargs_from_env_tls_verify_false_no_cert _______________________________________________________________

self = <tests.unit.utils_test.KwargsFromEnvTest testMethod=test_kwargs_from_env_tls_verify_false_no_cert>

    def test_kwargs_from_env_tls_verify_false_no_cert(self):
        temp_dir = tempfile.mkdtemp()
        cert_dir = os.path.join(temp_dir, '.docker')
>       shutil.copytree(TEST_CERT_DIR, cert_dir)

tests/unit/utils_test.py:224:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

src = '/home/tt/t/dockerpy-pypi-tests/docker-py-1.7.2/tests/unit/testdata/certs', dst = '/tmp/tmpc34nir8v/.docker', symlinks = False, ignore = None
copy_function = <function copy2 at 0x7f26369a9ea0>, ignore_dangling_symlinks = False

    def copytree(src, dst, symlinks=False, ignore=None, copy_function=copy2,
                 ignore_dangling_symlinks=False):
        """Recursively copy a directory tree.

        The destination directory must not already exist.
        If exception(s) occur, an Error is raised with a list of reasons.

        If the optional symlinks flag is true, symbolic links in the
        source tree result in symbolic links in the destination tree; if
        it is false, the contents of the files pointed to by symbolic
        links are copied. If the file pointed by the symlink doesn't
        exist, an exception will be added in the list of errors raised in
        an Error exception at the end of the copy process.

        You can set the optional ignore_dangling_symlinks flag to true if you
        want to silence this exception. Notice that this has no effect on
        platforms that don't support os.symlink.

        The optional ignore argument is a callable. If given, it
        is called with the `src` parameter, which is the directory
        being visited by copytree(), and `names` which is the list of
        `src` contents, as returned by os.listdir():

            callable(src, names) -> ignored_names

        Since copytree() is called recursively, the callable will be
        called once for each directory that is copied. It returns a
        list of names relative to the `src` directory that should
        not be copied.

        The optional copy_function argument is a callable that will be used
        to copy each file. It will be called with the source path and the
        destination path as arguments. By default, copy2() is used, but any
        function that supports the same signature (like copy()) can be used.

        """
>       names = os.listdir(src)
E       FileNotFoundError: [Errno 2] No such file or directory: '/home/tt/t/dockerpy-pypi-tests/docker-py-1.7.2/tests/unit/testdata/certs'

/usr/lib64/python3.5/shutil.py:303: FileNotFoundError
============================================================= 4 failed, 298 passed, 2 skipped, 1 pytest-warnings in 0.86 seconds =============================================================
```

--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -3,3 +3,4 @@ include requirements.txt
 include README.md
 include LICENSE
 recursive-include tests *.py
+recursive-include tests/unit/testdata *

